File: src/main/java/com/almox/model/entidades/Fornecedor.java
package com.almox.model.entidades;


import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "forn_fornecedor")
public class Fornecedor extends PessoaJuridica {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "forn_id")
    private Long id;

    @JsonIgnore
    @ManyToMany(fetch = FetchType.LAZY)
    private Set<Produto> produtosFornecidos;

    @Builder
    public Fornecedor(Long id, String razaoSocial, String cnpj, String nomeFantasia, Contato contato) {
        super(razaoSocial, cnpj, nomeFantasia, contato);
        this.id = id;
    }
}

File: frontend/src/model/fornecedor.ts
import { PessoaJurica } from './pessoa-juridica';
import { Contato } from './contato';

export interface Fornecedor extends PessoaJurica{
    id?: number;
    razaoSocial?:string;
    nomeFantasia?:string;
    rg?:string;
    cnpj?:string;
    contato?:Contato;

}
